app.rb

```diff
@@ -0,0 +1,94 @@
+require 'sinatra'
+require 'sinatra/activerecord'
+require 'active_record'
+require './models/user'
+require 'oauth2'
+require 'json'
+require 'sinatra/partial'
+require 'jwt'
+require 'sinatra/flash'
+require 'pry'
+require_relative 'authentificator'
+
+Dotenv.load
+register Sinatra::Partial
+enable :partial_underscores
+set :partial_template_engine, :erb
+enable :sessions
+
+LINK_USER_INFO = 'https://www.googleapis.com/userinfo/email?alt=json'.freeze
+NOT_AUTHORIZED_ERROR_MESSAGE = 'You are nor autorized'.freeze
+
+signing_key_path = File.expand_path('../app.rsa', __FILE__)
+verify_key_path  = File.expand_path('../app.rsa.pub', __FILE__)
+
+signing_key = File.open(signing_key_path) { |file| OpenSSL::PKey.read(file) }
+verify_key  = File.open(verify_key_path)  { |file| OpenSSL::PKey.read(file) }
+
+set :signing_key, signing_key
+set :verify_key,  verify_key
+
+get '/' do
+  @users = User.all
+
+  if authorized?
+    @encoded_token = JWT.encode(session[:access_token], settings.signing_key, 'RS256')
+    @decoded_token = JWT.decode(@encoded_token, settings.verify_key, true)
+  end
+
+  erb :index
+end
+
+get '/auth' do
+  redirect authentificator.authorize_url
+end
+
+get '/oauth2callback' do
+  autorize(params)
+
+  redirect '/'
+end
+
+get '/logout' do
+  session[:access_token] = nil
+
+  redirect '/'
+end
+
+post '/create' do
+  User.create(first_name: params[:username])
+
+  redirect '/'
+end
+
+post '/delete' do
+  begin
+    User.last.delete
```

Comment:
strange behavior with deleting the last user.
Okay, let it be so.
But you don't check if user is authorized. That's not secure.